File: src/FSharpVSPowerTools/TaskListCommentFilterProvider.cs
using System.ComponentModel.Composition;
using Microsoft.VisualStudio.Text.Editor;
using Microsoft.VisualStudio.Utilities;
using Microsoft.VisualStudio.Shell;
using FSharpVSPowerTools.TaskList;
using Microsoft.VisualStudio.Editor;
using FSharpVSPowerTools.ProjectSystem;
using Microsoft.VisualStudio.TextManager.Interop;

namespace FSharpVSPowerTools
{
    [Export(typeof(IVsTextViewCreationListener))]
    [ContentType("F#")]
    [TextViewRole(PredefinedTextViewRoles.Editable)]
    internal class TaskListCommentFilterProvider : IVsTextViewCreationListener
    {
        [Import]
        internal IVsEditorAdaptersFactoryService editorFactory = null;

        [Import(typeof(SVsServiceProvider))]
        internal System.IServiceProvider serviceProvider = null;

        [Import]
        internal OpenDocumentsTracker openDocsTracker = null;

        internal IWpfTextView textView;
        internal TaskListCommentFilter taskCommentFilter;

        public void VsTextViewCreated(IVsTextView textViewAdapter)
        {
            textView = editorFactory.GetWpfTextView(textViewAdapter);
            if (textView == null) return;

            var generalOptions = Setting.getGeneralOptions(serviceProvider);
            if (generalOptions == null || !generalOptions.TaskListCommentsEnabled) return;

            taskCommentFilter = new TaskListCommentFilter(textView, serviceProvider, openDocsTracker);
        }
    }
}

File: src/FSharpVSPowerTools.Pure/Commands/TaskListCommentFilterProvider.fs
module FSharpVSPowerTools.TaskListCommentFilterProvider

open System
open System.ComponentModel.Composition
open Microsoft.VisualStudio.Text
open Microsoft.VisualStudio.Text.Editor
open Microsoft.VisualStudio.Utilities
open Microsoft.VisualStudio.Shell
open FSharpVSPowerTools.TaskList
open Microsoft.VisualStudio.Editor
open FSharpVSPowerTools.ProjectSystem
open Microsoft.VisualStudio.TextManager.Interop


[<Export (typeof<IVsTextViewCreationListener>)>]
[<ContentType "F#">]
[<TextViewRole (PredefinedTextViewRoles.Editable)>]
type TaskListCommentFilterProvider [<ImportingConstructor>]
    ( [<Import (typeof<SVsServiceProvider>)>] 
        serviceProvider             :   IServiceProvider                ,
        editorFactory               :   IVsEditorAdaptersFactoryService ,
        textDocumentFactoryService  :   ITextDocumentFactoryService     ,
        taskListManager             :   TaskListManager                 ) =

    let mutable taskCommentFilter = Unchecked.defaultof<TaskListCommentFilter>

    interface IVsTextViewCreationListener with

        member __.VsTextViewCreated textViewAdapter =
            maybe {
                let! textView = editorFactory.TryGetWpfTextView textViewAdapter
                let! generalOptions = Setting.tryGetGeneralOptions serviceProvider
                if not generalOptions.TaskListCommentsEnabled then return! None else
                taskCommentFilter <- new TaskListCommentFilter 
                                        (   textView, serviceProvider, taskListManager, 
                                            OpenDocumentsTracker textDocumentFactoryService )
            } |> ignore
